constructs/base-function.ts

```diff
@@ -0,0 +1,94 @@
+import * as cdk from 'aws-cdk-lib';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambdaNodeJs from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import type { FullHandlerDefinition } from '../extract/extract-handlers';
+import { LambdaServiceProps } from './lambda-service';
+import { ISecurityGroup, IVpc, SubnetSelection } from 'aws-cdk-lib/aws-ec2';
+import { HandlerDefinition } from '../handlers/handler';
+
+export interface BaseFunctionProps<T extends HandlerDefinition> {
+	role: iam.IRole;
+	definition: FullHandlerDefinition<T>;
+	bundlingOptions?: lambdaNodeJs.BundlingOptions;
+	layers?: lambda.ILayerVersion[];
+	defaults?: LambdaServiceProps['defaults'];
+	vpc?: IVpc;
+	vpcSubnets?: SubnetSelection;
+	securityGroups?: ISecurityGroup[];
```

Comment:
I wonder if we can make this type better reflect that they need to be used together?

I'd prefer to avoid nested types, but 
```
network?: {
vpc: ...,
vpcSubnets: ...,
securityGroup: ...,
} ```
seems like the easiest way to do that.

Otherwise we have to have some kind of complex union type here, which I'm not sure is worth it?

Reply:
Looks like it's possible without nested types, the union type is actually very straightforward. I'm adding that.

Reply:
It did require nested types, but I think it better describes the API.